FrontendApartmentReservation/src/actions/index.jsx

```diff
@@ -12,75 +22,96 @@ var headers = {
+                });
+            }).catch((error) => console.warn(error));
     }).catch((error) => console.warn(error));
 }
 
 export const getAllAuthentication = () => (dispatch) => {
-    return fetch(`/api/authentications/`, {
+    return fetch(BACKEND_BASE_URI + `/api/authentication`, {
         method: "GET",
         headers: headers
     }).then(response => {
         if (!response.ok) {
             throw new Error("Bad response");
-          }
+        }
         response.json()
-        .then(data => {
-            dispatch({
-                type: GET_ALL_AUTHENTICATIONS,
-                payload: data
+            .then(data => {
+                dispatch({
+                    type: GET_ALL_AUTHENTICATIONS,
+                    payload: data
+                });
             });
-        });
     }).catch((error) => console.warn(error));
 }
 
 export const getAllUsers = () => (dispatch) => {
     console.log("indexas");
-    return fetch(`/api/users/`, {
+    return fetch(BACKEND_BASE_URI + `/api/profiles`, {
         method: "GET",
         headers: headers
     }).then(response => {
         if (!response.ok) {
             throw new Error("Bad response");
-          }
+        }
         response.json()
-        .then(data => {
-            dispatch({
-                type: GET_ALL_USERS,
-                payload: data
+            .then(data => {
+                dispatch({
+                    type: GET_ALL_USERS,
+                    payload: data
+                });
             });
-        });
     }).catch((error) => console.warn(error));
 }
 
 export const getAllTrips = () => (dispatch) => {
-    return fetch(`/api/authentications/`, {
+    return fetch(BACKEND_BASE_URI + `/api/travels`, {
         method: "GET",
         headers: headers
     }).then(response => {
         if (!response.ok) {
             throw new Error("Bad response");
-          }
+        }
         response.json()
-        .then(data => {
-            dispatch({
-                type: GET_ALL_TRIPS,
-                payload: data
+            .then(data => {
+                dispatch({
+                    type: GET_ALL_TRIPS,
+                    payload: data
+                });
             });
-        });
     }).catch((error) => console.warn(error));
 }
+
+export const signUpUser = (FirstName, LastName, Email, Password, Office) => (dispatch) => {
+    fetch(BACKEND_BASE_URI + "/api/users/profiles", {
+        method: "POST",
+        body: JSON.stringify({ FirstName, LastName, Email, Password, Office }),
+        headers,
+    }).then(response => {
+        if (response.status === 200) {
+            response.json().then(data => {
+                dispatch({
+                    type: SIGN_UP_USER,
+                    payload: data
+                }).then(() => {
+                    dispatch(push('/'))
+                });
+            });
+        }
+        else {
+            alert(i18n.t("SignUpError") + response.status);
```

Comment:
Nice job here too. Backend uses `_`s instead of capitalization, but we can allign on that.

You can also use `response.ErrorCode` directly as a key for translation.